### ISSUE TITLE
Non-standard naming in L-BFGS Learners (LogisticRegression, PoissonRegression)

### ISSUE THREAD

**Issue by rogancarr** [dotnet/machinelearning#2882]
In `LogisticRegression` and `PoissonRegression` (which use the same L-BFGS base), we have a parameter `IterationsToRemember` that refers to the number of gradients to accumulate in the history. While this terminology makes sense, it's not what we usually encounter in the field.

In the literature, we see this referred to as the "history size" (see e.g. [wikipedia](https://en.wikipedia.org/wiki/Limited-memory_BFGS)).

In the various toolkits that expose an L-BFGS solver, they use:
Scikit Learn: Doesn't expose it.
Spark: [NumberOfCorrections](https://spark.apache.org/docs/2.3.0/api/java/org/apache/spark/mllib/optimization/LBFGS.html)
TensorFlow: [num_correction_pairs](https://www.tensorflow.org/probability/api_docs/python/tfp/optimizer/lbfgs_minimize)
PyTorch: [history_size](https://pytorch.org/docs/stable/_modules/torch/optim/lbfgs.html)

I would vote for `HistorySize`, with the docs explaining what it is.

What do you all think? Any big feelings around this?

**Comment by rogancarr** [comment]
@shauheen @wschin @TomFinley @markusweimer  What do you think?

**Comment by Ivanidzo4ka** [comment]
Just to protect myself from pitchforks, before it was `MemorySize`. :D

**Comment by rogancarr** [comment]
Oh, I don't think `IterationsToRemember` is necessarily bad, just not standard :)

**Comment by TomFinley** [comment]
> Just to protect myself from pitchforks, before it was `MemorySize`. :D

I don't remember that. 😉

Jokes aside, yes totally sounds fine with me.

**Comment by rogancarr** [comment]
@TomFinley `HistorySize` or `IterationsToRemember`?

**Comment by shauheen** [comment]
`HistorySize` +1
